### PR TITLE
523 0box wallet system tests

### PR DESCRIPTION
**Pull request by Jayashsatolia403**
Added all 4 system tests of API endpoints for wallets.

### CODE REVIEW

File: tests/api_tests/0box_test.go [hunk 1]
```diff
@@ -190,6 +192,190 @@ func Test0Box(testSetup *testing.T) {
...
+		require.NoError(t, err)
+		require.Equal(t, 200, response.StatusCode(), "Response status code does not match expected. Output: [%v]", response.String())
+		require.NotNil(t, zboxWalletKeys)
+		require.NotEqual(t, 0, len(response.String()), "Response body is empty")
+	})
+
+	t.RunSequentially("Get wallet keys should return empty with wallet not present", func(t *test.SystemTest) {
+		teardown(t, firebaseToken.IdToken, zboxClient.DefaultPhoneNumber)
+		csrfToken := createCsrfToken(t, zboxClient.DefaultPhoneNumber)
+
```
Comment (stewartie4) [inline tests/api_tests/0box_test.go:223]: general: lint issues need resolved

File: tests/api_tests/0box_test.go [hunk 2]
```diff
@@ -200,6 +386,7 @@ func teardown(t *test.SystemTest, idToken, phoneNumber string) {
 	if wallets != nil {
 		t.Logf("Found [%v] existing wallets for [%v]", len(wallets), phoneNumber)
 		for _, wallet := range wallets {
+			fmt.Println(wallet.WalletId)
 			message, response, err := zboxClient.DeleteWallet(t, wallet.WalletId, idToken, csrfToken, phoneNumber)
 			println(message, response, err)
 		}
```
Comment (stewartie4) [inline tests/api_tests/0box_test.go:392]: Why do we need to add these jpgs to the repo?
the other tests generate a file, this also avoids file permission issues

File: tests/api_tests/0box_test.go [hunk 3]
```diff
@@ -190,6 +192,190 @@ func Test0Box(testSetup *testing.T) {
...
+		response, err = zboxClient.ContactWallet(t, reqBody, firebaseToken.IdToken, csrfToken, zboxClient.DefaultPhoneNumber)
+
+		_ = json.Unmarshal([]byte(response.String()), &cr)
+
+		require.NoError(t, err)
+		require.Equal(t, 200, response.StatusCode(), "Response status code does not match expected. Output: [%v]", response.String())
+		require.Equal(t, 1, len(cr.Data), "Response data does not match expected. Output: [%v]", response.String())
+	})
+
+	t.RunSequentially("Contact Wallet should work with for multiple users", func(t *test.SystemTest) {
```
Comment (stewartie4) [inline tests/api_tests/0box_test.go:345]: negative scenarios could be added eg.
field validation (invalid, missing field, length check etc.)
404 not found 
etc.

**Review by stewartie4 — CHANGES_REQUESTED** [review]

**Review by stewartie4 — COMMENTED** [review]
Great set of tests, just some minor changes

**Comment by Jayashsatolia403** [comment]
Sure, I will fix the lint issues and add some negative scenarios. Thanks for the comment.

**Comment by Jayashsatolia403** [comment]
Fixed and Updated the branch.

**Pull request closed without merging**